migration: Delete namespaces obsolete since v0.20

Zash opened the pull request:
<!-- Choose you PR title carefully as it will be used as the entry in the changelog! -->

> [!warning]
> **This is public repository, ensure not to disclose:**
> - [ ] personal data beyond what is necessary for interacting with this pull request
> - [ ] business confidential information, such as customer names

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       <!-- This PR adds a new feature -->
- [ ] kind/improvement   <!-- This PR changes an existing feature -->
- [ ] kind/deprecation   <!-- This PR removes an existing feature -->
- [ ] kind/documentation <!-- This PR contains documentation -->
- [x] kind/clean-up      <!-- This PR cleans up technical debt -->
- [ ] kind/bug           <!-- This PR fixes a bug -->
- [ ] kind/other         <!-- This PR does something else -->

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [ ] kind/admin-change  <!-- This PR introduces an admin facing change, add "Platform Administrator notice" section -->
- [ ] kind/dev-change    <!-- This PR introduces a dev facing change, add "Application Developer notice" section -->
- [ ] kind/security      <!-- This PR introduces a critical security fix, add "Security notice" section -->
- [ ] [kind/adr]()       <!-- This PR implements an ADR, add the link -->

<!-- Uncomment the additional sections that applies. -->

<!-- Additional information to be added in the release notes
### Release notes
...
-->

<!-- Additional information with kind/admin-change
### Platform Administrator notice
...
-->

<!-- Add additional information with kind/dev-change
### Application Developer notice
...
-->

<!-- Add additional information with kind/security
### Security notice
...
-->

### What does this PR do / why do we need this PR?

<!-- Add description of the change -->
**_Removes_** namespaces that have been obsolete since v0.20, namely:

- `elastic-system`
- `influxdb-prometheus`

Rollback not implemented because the namespaces have been obsolete so long, plus this wouldn't restore any resources in the deleted namespaces anyway, so restoring them on their own seems of little value.

<!-- Add all issues that are fixed by this PR, use "Part of" instead of "Fixes" if you want to keep issues open. -->
- Fixes #1585

#### Additional information to reviewers

#### Screenshots

#### Testing done

1. ran migration against cluster without the namespaces in question, script should handle this gracefully
2. created both namespaces
3. ran migration again, observed that it claimed to have deleted the namespaces
4. observed that `kubectl get namespace` lists neither namespace, in either cluster

#### Checklist

<!-- This section is not added to the changelog or release notes, it is to help you as a contributor and reviewers. -->

- [x] Proper commit message prefix on all commits
  <!-- Example of commit message prefixes:
    - all: changes to multiple areas
    - apps: changes to application running in all clusters
    - apps sc: changes to applications running in the service cluster
    - apps wc: changes to applications running in the workload cluster
    - bin: changes to management binaries or scripts
    - config: changes to configuration
    - docs: changes to documentation
    - tests: changes to tests
    - pipeline: changes to the pipeline
    - release: release related
  -->
- Change checks:
  - [ ] The change is transparent
  - [ ] The change is disruptive
  - [ ] The change requires no migration steps
  - [x] The change requires migration steps
- Metrics checks:
  - [ ] The metrics are still exposed and present in Grafana after the change
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [ ] The logs do not show any errors after the change
- Network Policy checks:
  - [ ] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packages in the `NetworkPolicy Dashboard`
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy

linus-astrom commented on migration/v0.34/apply/30-delete-obsolete-ns.sh:
```diff
@@ -0,0 +1,33 @@
+    for CLUSTER in sc wc; do
+      for NS in elastic-system influxdb-prometheus; do
+        if kubectl_do $CLUSTER get namespace $NS >/dev/null 2>/dev/null; then
+          # Sensible to use warn for potentially destructive actions?
```
You should probably remove the comments at line 11 and 15.
They seem more like questions for people, then comments meant to show how a function in the code works.

Zash replied:
Done.

linus-astrom commented:
You should update [manual method](https://github.com/elastisys/compliantkubernetes-apps/tree/main/migration/v0.34#manual-method) to include your migration step.

Zash commented:
> You should update the [manual method](https://github.com/elastisys/compliantkubernetes-apps/tree/main/migration/v0.34#manual-method) to include your migration step.

Added.

Also added a check in the `prepare` step checking if there are any resources left in the namespaces.

linus-astrom commented on migration/v0.34/README.md:
````diff
@@ -94,6 +94,22 @@ As with all scripts in this repository `CK8S_CONFIG_PATH` is expected to be set.
+
+    ``` bash
+    ./bin/ck8s ops kubectl sc get all -n elastic-system
+    ./bin/ck8s ops kubectl sc get all -n influxdb-prometheus
````
this (migration file)[https://github.com/elastisys/compliantkubernetes-apps/tree/d0d207a532078d64d084b1398b7b0401cad657d5/migration/v0.33#prepare-upgrade---non-disruptive] shows it better manual method is more like do each individual prepare/apply file instead of doing the whole prepare/apply file  that the automatic method does 
```suggestion
    ./migration/v0.34/prepare/30-delete-obsolete-ns.sh
```

linus-astrom commented on migration/v0.34/README.md:
````diff
@@ -113,6 +129,13 @@ As with all scripts in this repository `CK8S_CONFIG_PATH` is expected to be set.
+
+    ```bash
+    ./bin/ck8s ops kubectl sc delete ns elastic-system
+    ./bin/ck8s ops kubectl sc delete ns influxdb-prometheus
````
same as above comment but also move it to the right order to around [here](https://github.com/elastisys/compliantkubernetes-apps/blob/d0d207a532078d64d084b1398b7b0401cad657d5/migration/v0.34/README.md?plain=1#L152)

Zash replied:
Like 68655f32348181ebd71e10f087cdecb496239297 then

linus-astrom replied:
exactly

linus-astrom commented on migration/v0.34/README.md:
````diff
@@ -127,6 +142,12 @@ As with all scripts in this repository `CK8S_CONFIG_PATH` is expected to be set.
+1. Delete obsolete namespaces:
+
+    ```bash
+    ./migration/v0.34/apply/30-delete-obsolete-ns.sh
````
```suggestion
    ./migration/v0.34/apply/30-delete-obsolete-ns.sh execute
```

aarnq reviewed (approved):
Nice use of a prepare snippet. :+1:

Zash commented on migration/v0.34/prepare/30-delete-obsolete-ns.sh:
```diff
@@ -0,0 +1,20 @@
+# shellcheck source=scripts/migration/lib.sh
+source "${ROOT}/scripts/migration/lib.sh"
+
+log_info "no operation: this is a template"
```
👀 
```suggestion
```

OlleLarsson commented on migration/v0.34/prepare/30-delete-obsolete-ns.sh:
```diff
@@ -0,0 +1,20 @@
+# shellcheck source=scripts/migration/lib.sh
+source "${ROOT}/scripts/migration/lib.sh"
+
+log_info "no operation: this is a template"
```
```suggestion
```

Zash replied:
gone
